libsolidity/formal/Predicate.cpp

```diff
@@ -446,3 +459,26 @@ bool Predicate::fillArray(smtutil::Expression const& _expr, vector<string>& _arr
 
 	solAssert(false, "");
 }
+
+map<string, optional<string>> Predicate::readTxVars(smtutil::Expression const& _tx) const
+{
+	map<string, TypePointer> const txVars{
+		{"block.chainid", TypeProvider::uint256()},
+		{"block.coinbase", TypeProvider::address()},
+		{"block.difficulty", TypeProvider::uint256()},
+		{"block.gaslimit", TypeProvider::uint256()},
+		{"block.number", TypeProvider::uint256()},
+		{"block.timestamp", TypeProvider::uint256()},
+		{"blockhash", TypeProvider::array(DataLocation::Memory, TypeProvider::uint256())},
+		{"msg.data", TypeProvider::bytesMemory()},
+		{"msg.sender", TypeProvider::address()},
+		{"msg.sig", TypeProvider::uint256()},
+		{"msg.value", TypeProvider::uint256()},
+		{"tx.gasprice", TypeProvider::uint256()},
+		{"tx.origin", TypeProvider::address()}
+	};
```

Comment:
Not too happy about the duplication of these names, but not sure if extracting would make it worse. I think it's fine like this for now.